lib/rules/web/ocm_console/cluster_idp.xyaml

```diff
@@ -0,0 +1,680 @@
+# --- the elements and actions about the idp configuration
+check_idp_hint:
+  elements:
+  - selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__icon']
+    timeout: 100
+  - selector:
+      xpath: //div[@id='idpHint']/h4[@class='pf-c-alert__title' and contains(text(),'Missing Identity Providers')]
+  - selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__description' and contains(text(),'Identity Providers determine how users log into the cluster')]
+  action: check_add_oauth_configuration_button
+check_add_oauth_configuration_button:
+  element: &oauth_configuration_button
+    selector:
+      xpath: //div[@id='idpHint']/div[@class='pf-c-alert__description']/button[text()='Add OAuth Configuration']
```

Comment:
Simple the rule to `//div[@id='idpHint']//button[text()='Add OAuth Configuration']`

Reply:
@yasun1  The part of "div[@class='pf-c-alert__description']"  indicates the button is under the alert message field, I think it is not a redundancy xpath part and we should keep it.